fdbserver/RestoreWorkerInterface.actor.h

```diff
@@ -0,0 +1,544 @@
+/*
+ * RestoreWorkerInterface.actor.h
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// This file declare and define the interface for RestoreWorker and restore roles
+// which are RestoreMaster, RestoreLoader, and RestoreApplier
+
+
+#pragma once
+#if defined(NO_INTELLISENSE) && !defined(FDBSERVER_RESTORE_WORKER_INTERFACE_ACTOR_G_H)
+	#define FDBSERVER_RESTORE_WORKER_INTERFACE_ACTOR_G_H
+	#include "fdbserver/RestoreWorkerInterface.actor.g.h"
+#elif !defined(FDBSERVER_RESTORE_WORKER_INTERFACE_ACTOR_H)
+	#define FDBSERVER_RESTORE_WORKER_INTERFACE_ACTOR_H
+
+
+#include <sstream>
+#include "flow/Stats.h"
+#include "fdbrpc/fdbrpc.h"
+#include "fdbrpc/Locality.h"
+#include "fdbclient/FDBTypes.h"
+#include "fdbclient/CommitTransaction.h"
+#include "fdbserver/CoordinationInterface.h"
+#include "fdbserver/Knobs.h"
+#include "fdbserver/RestoreUtil.h"
+
+#include "flow/actorcompiler.h" // has to be last include
+
+#define DUMPTOKEN( name ) TraceEvent("DumpToken", recruited.id()).detail("Name", #name).detail("Token", name.getEndpoint().token)
+
+class RestoreConfig;
+
+struct RestoreCommonReply;
+struct RestoreRecruitRoleRequest;
+struct RestoreSysInfoRequest;
+struct RestoreLoadFileRequest;
+struct RestoreVersionBatchRequest;
+struct RestoreSendMutationVectorVersionedRequest;
+struct RestoreSetApplierKeyRangeVectorRequest;
+struct RestoreSysInfo;
+struct RestoreApplierInterface;
+
+
+struct RestoreSysInfo {
+	std::map<UID, RestoreApplierInterface> appliers;
+
+	RestoreSysInfo() = default;
+	explicit RestoreSysInfo(std::map<UID, RestoreApplierInterface> appliers) : appliers(appliers) {}
```

Comment:
I thought about this. The reason why I keep it a more generic name is because I think it may include more information in the future.

How about keeping the `RestoreSysInfo` name for now. If it turns out this struct only has appliers in the final version, we will rename it to a more specific name, e.g., AppliersMap. (I will add a TODO above this struct to remind me about this.)